### ISSUE TITLE
Custom cmake targets vsg_add_target_xxx are not present in out of source builds

### ISSUE THREAD

**Issue by rhabacker** [vsg-dev/VulkanSceneGraph#568]
**Describe the bug**
When building vsg on a local linux host with an out of source build dir the custom cmake targets `cppcheck`, `clang-format`, `uninstall`, `docs` and `clobber` are not present.

**To Reproduce**
1. install vulkan development package 
2. Fetch VulkanSceneGraph sources from github into `~/src/VulkanSceneGraph`
3. run 
```
mkdir ~/src/VulkanSceneGraph-build
cd ~/src/VulkanSceneGraph-build
cmake ../VulkanSceneGraph
make cppcheck
make: *** No rule to create „cppcheck“ 

**Expected behavior**
The mentioned cmake targets should be available in an out of source build

**Desktop (please complete the following information):**
 - OS: openSUSE_Leap_15.4

**Comment by robertosfield** [comment]
The lack of cppcheck and clang-format on an out of source build is not ideal, but for VSG-1.0.0 I don't see it as an issue worth prioritizing.

The issue with uninstall, clobber and docs is troubling and would be good to fix before 1.0.0 goes out. I am focusing on client work today, tomorrow I have cleared for release work so can look into it then.

**Comment by rhabacker** [comment]
A consequential problem with an out of source build of vsg is that the cmake file uninstall.cmake required for the uninstall target is not installed, leading to build system errors when running the uninstall target in vsg dependent projects. 
This is fixed by the first commit of pr #566.

**Comment by robertosfield** [comment]
I have tried out the first commit from #566 and this adds installation of the uninstall.cmake into lib/cmake/vsg but it doesn't add the make target.  I'll review the changes more widely to see if we can get this working with significant more changes.

**Comment by rhabacker** [comment]
The target is added by the last commit - maybe I should change the order from 1,2,3 to 2,3,1

**Comment by robertosfield** [comment]
It's getting more complicated having to juggle commits, I have merged the PR's as branch so will test everything, then if that goes OK will look at merging with VSG master, then do a call for testing, if that's OK merge with 1.0.  It's something I'm reluctant to do as it's quite a few changes to merge close to the release, but it looks like the easiest way to resolve the out of source build.

**Comment by robertosfield** [comment]
I have merged the #566 with VSG master which resolves this issue for me.  Waiting on community testing to report things are OK with cross platform build before merging with VulkanSceneGraph-1.0 branch.

**Comment by rhabacker** [comment]
I'll take a look today.
